tracer/test/Datadog.Trace.Tests/Propagators/MultiSpanContextPropagatorTests.cs

```diff
@@ -200,7 +221,7 @@ public void Extract_W3C_IHeadersCollection_traceparent_tracestate()
                            RawSpanId = "000000003ade68b1",
                            SamplingPriority = SamplingPriorityValues.UserKeep,
                            Origin = "rum",
-                           PropagatedTags = "_dd.p.dm=-4,_dd.p.usr.id=12345",
+                           PropagatedTags = PropagatedTagsCollection,
```

Comment:
Does this test case pass? PropagatedTagsCollection is composed of tags "_dd.p.key1=value1,_dd.p.key2=value2" but the test should be expecting a TraceTagCollection composed of tags "_dd.p.dm=-4,_dd.p.usr.id=12345"

Reply:
Ok I figured it out. FluentAssertions was comparing the expected `SpanContextMock` with the actual `SpanContext` and doing a structural equality test, so going through the properties and comparing the properties whose names matched. However, it incorrectly says the PropagatedTags are equal when we should be reporting the error.

The issue is that the `TraceTagCollection` object comparison is also done by structural equality, but the only public property to compare is `public int Count { get; }`. Since the Count returns 2 for each object, the `TraceTagCollection` object comparison was considered successful.

To fix this, there's two options that should work:
1. Add a new public property `public List<KeyValuePair<string, string>>? Tags { get; }` on the TraceTagCollection type (or maybe convert the tags to a readonly collection)
2. Update every SpanContextMock to look like this:
```
result.Should()
      .NotBeNull()
      .And
      .BeEquivalentTo(
           new SpanContextMock
           {
               TraceId = 123456789,
               SpanId = 987654321,
               RawTraceId = "000000000000000000000000075bcd15",
               RawSpanId = "000000003ade68b1",
               SamplingPriority = SamplingPriorityValues.UserKeep,
               Origin = "rum",
               PropagatedTags = PropagatedTagsCollection,
               Parent = null,
               ParentId = null,
           },
           opt => opt.Using<TraceTagCollection>(ctx => ctx.Subject.ToArray().Should().BeEquivalentTo(ctx.Expectation.ToArray()))
                     .WhenTypeIs<TraceTagCollection>());
```

Reply:
Oh, wow, good catch!

Reply:
Fixed. I went with option 1 and added a `Tags` property to `TraceTagCollection`:
```
public IReadOnlyList<KeyValuePair<string, string>> Tags => ToArray();
```
then this started to fail. Luckily this was the only test with this issue 😅